### PR TITLE
Add stricter validation for some ingress annotations

### PR DESCRIPTION
**Pull request by mikestephen**
### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: internal/configs/parsing_helpers.go [hunk 1]
```diff
@@ -225,6 +225,19 @@ func ParseSize(s string) (string, error) {
 	return "", errors.New("Invalid size string")
 }
 
+// https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffers
+var proxyBuffersRegexp = regexp.MustCompile(`^\d+ \d+[kKmM]?$`)
+
+// ParseProxyBuffersSpec ensures that the string value is a valid proxy buffer spec
+func ParseProxyBuffersSpec(s string) (string, error) {
```
Comment (pleshakov) [inline internal/configs/parsing_helpers.go:232]: would it help to have a unit test for this func?

**Review by pleshakov — APPROVED** [review]

File: internal/configs/parsing_helpers.go (hunk 1 above)
Comment (mikestephen) [reply internal/configs/parsing_helpers.go:232]: Thanks, I'd forgotten, added one now.

File: internal/configs/parsing_helpers_test.go [hunk 2]
```diff
@@ -423,6 +423,26 @@ func TestParseSize(t *testing.T) {
 	}
 }
 
+func TestParseProxyBuffersSpec(t *testing.T) {
+	var testsWithValidInput = []string{"1 1k", "10 24k", "2 2K", "6 3m", "128 3M"}
+	var invalidInput = []string{"-1", "-6 2k", "", "blah", "16k", "10M", "2 4g", "3 4G"}
+	for _, test := range testsWithValidInput {
+		result, err := ParseProxyBuffersSpec(test)
+		if err != nil {
+			t.Errorf("TestParseProxyBuffersSpec(%q) returned an error for valid input", test)
```
Comment (Dean-Coakley) [inline internal/configs/parsing_helpers_test.go:432]: I think the test error should match the func name, not the test func name. (Several similar cases in a few places in this file)
```suggestion
			t.Errorf("ParseProxyBuffersSpec(%q) returned an error for valid input", test)
```

**Review by Dean-Coakley — APPROVED** [review]

File: internal/configs/parsing_helpers_test.go [hunk 3]
```diff
@@ -423,6 +423,26 @@ func TestParseSize(t *testing.T) {
...
+func TestParseProxyBuffersSpec(t *testing.T) {
+	var testsWithValidInput = []string{"1 1k", "10 24k", "2 2K", "6 3m", "128 3M"}
+	var invalidInput = []string{"-1", "-6 2k", "", "blah", "16k", "10M", "2 4g", "3 4G"}
+	for _, test := range testsWithValidInput {
+		result, err := ParseProxyBuffersSpec(test)
+		if err != nil {
+			t.Errorf("ParseProxyBuffersSpec(%q) returned an error for valid input", test)
+		}
+		if test != result {
+			t.Errorf("TestParseProxyBuffersSpec(%q) returned %q expected %q", test, result, test)
```
Comment (Dean-Coakley) [inline internal/configs/parsing_helpers_test.go:435]:
```suggestion
			t.Errorf("ParseProxyBuffersSpec(%q) returned %q expected %q", test, result, test)
```

File: internal/configs/parsing_helpers_test.go [hunk 4]
```diff
@@ -423,6 +423,26 @@ func TestParseSize(t *testing.T) {
...
+		}
+	}
+	for _, test := range invalidInput {
+		result, err := ParseProxyBuffersSpec(test)
+		if err == nil {
+			t.Errorf("TestParseProxyBuffersSpec(%q) didn't return error. Returned: %q", test, result)
```
Comment (Dean-Coakley) [inline internal/configs/parsing_helpers_test.go:441]:
```suggestion
			t.Errorf("ParseProxyBuffersSpec(%q) didn't return error. Returned: %q", test, result)
```

**Review by vepatel — APPROVED** [review]

**Pull request merged**